hackathon_site/event/views.py

```diff
@@ -226,3 +227,43 @@ class TeamListView(mixins.ListModelMixin, generics.GenericAPIView):
 
     def get(self, request, *args, **kwargs):
         return self.list(request, *args, **kwargs)
+
+
+class MultipleFieldLookupMixin:
+    """
+    Apply this mixin to any view or viewset to get multiple field filtering
+    based on a `lookup_fields` attribute, instead of the default single field filtering.
+    """
+
+    def get_object(self, *args, **kwargs):
+        queryset = self.get_queryset()  # Get the base queryset
+        queryset = self.filter_queryset(queryset)  # Apply any filter backends
+        filter = {}
+        for field in self.lookup_fields:
+            if self.kwargs[field]:  # Ignore empty fields.
+                filter[field] = self.kwargs[field]
+        obj = get_object_or_404(queryset, **filter)  # Lookup the object
+        self.check_object_permissions(self.request, obj)
+        return obj
+
+
+class TeamCodeView(MultipleFieldLookupMixin, generics.GenericAPIView):
```

Comment:
Take a look at DRFs [RetrieveModelMixin](https://www.django-rest-framework.org/api-guide/generic-views/#retrievemodelmixin) - It should do what we need by just setting the lookup field. The get method then just needs to be:

```python
def get(self, request, *args, **kwargs):
    return self.retrieve(request, *args, **kwargs)
```

Take a look at the `CurrentUserAPIView` for an example of how to use it (you don't need the `get_object` method that's in there, that one had custom behaviour to get the object from the request user rather than a URL kwarg)

Reply:
Can you move this to `event/api_views.py`? That's something I should have caught with all the other API views we've done as well (IEEE-88 is a ticket to move all the other ones to where they should be).

Reply:
```suggestion
class TeamDetailView(mixins.RetrieveModelMixin, generics.GenericAPIView):
```

Calling things `<model>DetailView` and `<model>ListView` is a common convention. Usually they each have separate GET methods (detail view returns a single instance while list view returns a list), the detail view has PATCH (modify) and DELETE methods, and the list view has a POST (create) method

Reply:
Because in the ticket, it says we need to use team_code as the condition to retrieve the data. I have tested by using  RetrieveModelMixin, but it uses <int:pk> as the default condition and the solution I found to use team_code instead of <int:pk> as the query condition is to use MultipleFieldLookupMixin. @grahamhoyes

Reply:
Makes sense! I think you did right by the ticket, I'm questioning if the ticket itself is correct in what it's asking to do. There's some threads on slack discussing it too